engine/src/main/java/com/redhat/cloud/notifications/db/repositories/TemplateRepository.java

```diff
@@ -124,33 +124,56 @@ private InstantEmailTemplate loadDefaultEmailTemplate() throws Exception {
      * Integration templates are more generic templates directly targeted at Integrations
      * other than email, like Splunk, Slack or Teams via Camel. See {@link IntegrationTemplate}
      * for more details.
-     * @param appName Application name the template applies to. Can be null.
+     * @param appId Application name for the template applies to. Can be null.
+     * @param eventTypeId Event type Id for the template applies to. Can be null.
      * @param orgId The organization id for templates that are organization specific. Need to be of Kind ORG
      * @param templateKind Kind of template requested. If it does not exist, Kind DEFAULT is returned or Optional.empty()
      * @param integrationType Type of integration requested. E.g. 'slack', 'teams' or 'splunk'
      * @return IntegrationTemplate with potential fallback or Optional.empty() if there is not even a default template.
      */
-    public Optional<IntegrationTemplate> findIntegrationTemplate(String appName,
+    public Optional<IntegrationTemplate> findIntegrationTemplate(UUID appId,
```

Comment:
This is not your responsibility but I really really don't like this method (old or new one, doesn't matter). This is such a mess from a maintenance perspective...